### PR TITLE
Pass explicit font paths to fontspec in backend_pgf.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

xref #10307.  @laborleben Can you check whether this works for you?  Seems better than continuously posting patches :)

Basically
1. we already have a font fonding mechanism and should just reuse it in backend_pgf.
2. we sometimes find fonts in nonstandard locations and may need to tell tex exactly where to find them.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by laborleben** [comment]
Hi, thanks a lot for the work. This time it worked like a charm. I tried different parameters and the code worked every time. I'm looking forward to see your improvements merged upstream.

If you need further testing on macOS, please let me know.

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -77,23 +58,20 @@ def _parse_lualatex_version(output):
...
-            else:
-                pass  # no fonts found, fallback to LaTeX defaule
+        families = ["serif", "sans\\-serif", "monospace"]
+        commands = ["setmainfont", "setsansfont", "setmonofont"]
+        for family, command in zip(families, commands):
+            # 1) Forward slashes also work on Windows, so don't mess with
+            # backslashes.  2) The dirname needs to include a separator.
+            dirname = Path(fm.findfont(family)).parent.as_posix() + "/"
+            latex_fontspec.append(
+                r"\%s{%s}[Path=%s]" % (command, basename, dirname))
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pgf.py]: `basename` seems undefined. Apparently an untested code path :frowning:

File: lib/matplotlib/backends/backend_pgf.py [hunk 2]
```diff
@@ -77,23 +58,20 @@ def _parse_lualatex_version(output):
 def get_fontspec():
     """Build fontspec preamble from rc."""
     latex_fontspec = []
-    texcommand = get_texcommand()
+    texcommand = rcParams["pgf.texsystem"]
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_pgf.py]: Do we explicitly not want the correction and fallback mechanism to xelatex, which was implemented in `get_texcommand`?

Comment (anntzer) [reply lib/matplotlib/backends/backend_pgf.py]: That does already get checked during standard rcparam validation.

**Comment by anntzer** [comment]
I'll skip the bonus points.

**Pull request merged**